05_run_ocp.sh

```diff
@@ -106,7 +106,23 @@ ssh -o "StrictHostKeyChecking=no" core@$IP sudo modprobe iscsi_tcp
 # doesn't exist on the host
 ssh -o "StrictHostKeyChecking=no" core@$IP sudo mkdir /run/httpd
 
+# Internal dnsmasq should reserve IP addresses for each master
+cp -rf ironic/dnsmasq.conf ~stack/dnsmasq.conf
+for i in 0 1 2; do 
+  NODE_MAC=$(cat ~stack/ironic_nodes.json | jq -r ".nodes[${i}].ports[0].address")
+  NODE_IP="172.22.0.2${i}"
+  HOSTNAME="${CLUSTER_NAME}-etcd-${i}.${BASE_DOMAIN}"
+  # Make sure internal dnsmasq would assign an expected IP
+  echo "dhcp-host=${NODE_MAC},${HOSTNAME},${NODE_IP}" >> ~stack/dnsmasq.conf
+  # Reconfigure "external" dnsmasq
+  echo "${NODE_IP} ${HOSTNAME} ${CLUSTER_NAME}-api.${BASE_DOMAIN}" | sudo tee -a /etc/hosts.openshift
+  echo "srv-host=_etcd-server-ssl._tcp.${CLUSTER_NAME}.${BASE_DOMAIN},${HOSTNAME},2380,0,0" | sudo tee -a /etc/NetworkManager/dnsmasq.d/openshift.conf
```

Comment:
If I'm not mistaken, the etcd clustering resolution is only needed by the nodes themselves to cluster up. For this reason, I think it is better that you put DNS SRV records in the libvirt network definition like I do for A records in:
https://github.com/metalkube/dev-scripts/pull/38/files (network.xml.j2)

You can see an example of the SRV record in https://libvirt.org/formatnetwork.html#elementsAddress

It is for this reason that I am triyng to move the nodes from the 'default' libvirt node to a network I called 'baremetal'.

Reply:
In that libvirt section there's also the way how to make the host mac -> IP assignments you do here.

Reply:
The goal is for this baremetal network to eventually look and feel exactly like we expect the real baremetal network that will be used for the cluster networking.